vpr/src/place/initial_placement.cpp

```diff
@@ -166,63 +371,71 @@ static int try_place_macro(int itype, int ipos, int isub_tile, t_pl_macro pl_mac
     return (macro_placed);
 }
 
-static void initial_placement_pl_macros(int macros_max_num_tries, std::vector<std::vector<int>>& free_locations, const std::vector<t_pl_macro>& sorted_macros) {
+static void place_macro(int macros_max_num_tries, t_pl_macro pl_macro) {
```

Comment:
I think you can merge the macro and non-macro cases by making a macro on the fly with only one block in it.  Should reduce code duplication by only having one code path.

Reply:
Also merge place_block and place_macro.